mongodb/src/main/java/org/geotools/data/mongodb/MongoDataStore.java

```diff
@@ -198,8 +198,11 @@ public void createSchema(SimpleFeatureType incoming) throws IOException {
         incoming.getUserData().put(KEY_collection, incoming.getTypeName());
         
         // Collection needs to exist (with index) so that it's returned with createTypeNames()
-        dataStoreDB.createCollection(incoming.getTypeName(), new BasicDBObject())
-                .ensureIndex(new BasicDBObject(geometryMapping, "2dsphere"));
+        DBCollection collection = dataStoreDB.getCollection(incoming.getTypeName());
+        if (collection == null) {
+        	dataStoreDB.createCollection(incoming.getTypeName(), new BasicDBObject());
```

Comment:
I would expect this to be null as we are creating a new collection at this time.
If a collection already exists you may want to be using alter updateSchema??